### PR TITLE
Separate processor from web api in the codebase

### PR DESCRIPTION
**Pull request by semtexzv**
- Separate webapi and processor to different modules
- Update docs to run them separately
- Will require changes to `e2e-deploy`

### CODE REVIEW

**Comment by kgaikwad** [comment]
@semtexzv,
Just for reference - 
adding ticket link here - https://issues.redhat.com/browse/RHIROS-62. 
We will see after your changes, will it be worth adding this ticket or not. Accordingly, I will resume work on it.
Currently I have fixed [app_context](https://github.com/RedHatInsights/ros-backend/blob/main/ros/processors/inventory_events_consumer.py#L76) by importing `app` from ros.app inside `inventory_events_consumer.py`.
[PR](https://github.com/RedHatInsights/ros-backend/pull/23) is merged now.

**Comment by semtexzv** [comment]
Needs https://github.com/RedHatInsights/e2e-deploy/pull/2754 to work properly on CI

File: ros/lib/app.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from flask import Flask
+from ros.lib.models import db
+from ros.lib.config import DB_URI
+
+# Since we're using flask_sqlalchemy, we must create the flask app in both processor and web api
+app = Flask(__name__)
+# Initalize database connection
+app.config['SQLALCHEMY_DATABASE_URI'] = DB_URI
+db.init_app(app)
```
Comment (apuntamb) [inline ros/lib/app.py:9]: should `app.py` also live inside `lib`?

Comment (semtexzv) [reply ros/lib/app.py:9]: It is. I modified it to use local imports

File: ros/app.py [hunk 2]
```diff
@@ -2,9 +2,9 @@
...
 from flask_restful import Api
 from flask_cors import CORS
-from ros.models import db
-from ros.config import DB_URI
-from ros.resources.routes import initialize_routes
+from ros.lib.models import db
+from ros.lib.config import DB_URI
+from ros.api.routes import initialize_routes
 
 app = Flask(__name__)
```
Comment (patilsuraj767) [inline ros/app.py:9]: I think we can delete this `app.py` file.  Once this PR is merged it will be a dead code.

File: scripts/docker-compose.yml [hunk 3]
```diff
@@ -100,17 +100,50 @@ services:
...
+    <<: *inventory
+    command: "make upgrade_db run_inv_web_service"
+    depends_on:
+      - insights-inventory-mq
+    ports:
+      - 8081:8080
+
   puptoo:
-    image: quay.io/cloudservices/insights-puptoo:latest
+    image: quay.io/cloudservices/insights-puptoo:stable
```
Comment (patilsuraj767) [inline scripts/docker-compose.yml:114]: If the new sample report is working with puptoo, then can we keep it latest?

**Review by patilsuraj767 — CHANGES_REQUESTED** [review]
Thanks, @semtexzv Overall PR looks nice, I have tested it in local setup it working without any issue. 
Below added two minor queries.

File: ros/app.py (hunk 2 above)
Comment (semtexzv) [reply ros/app.py:9]: You're right, this is a merge error.

File: scripts/docker-compose.yml (hunk 3 above)
Comment (semtexzv) [reply scripts/docker-compose.yml:114]: Fixed

**Comment by kgaikwad** [comment]
@patilsuraj767, @apuntamb any additional comments/suggestions from your side?

**Comment by apuntamb** [comment]
> @patilsuraj767, @apuntamb any additional comments/suggestions from your side?

I haven't tested it on local, but I guess @patilsuraj767 has. If it's working as expected, it would be good to merge. No additional comments codewise from my side. :+1:

**Review by kgaikwad — APPROVED** [review]
@semtexzv,
LGTM :+1:  :ok_hand: 

Verified on local with below steps:
Started dev setup -
1. docker-compose up
2. pipenv run python3 -m ros.api.main
3. pipenv run python3 -m ros.processor.main

Tried scenarios -
1. Using make commands, uploaded data & verified using DB queries.
2. DELETE event
3. API request /systems

**Comment by kgaikwad** [comment]
@patilsuraj767 @apuntamb, merging these changes to unblock other PRs.
In case any further suggestions/comments, I think it would be good to address in separate PR.

Thank you @semtexzv!

**Pull request merged**